packages/amazon-cognito-identity-js/__mocks__/mocks.js

```diff
@@ -0,0 +1,72 @@
+import Client from '../src/Client';
+import AuthenticationHelper from '../src/AuthenticationHelper';
+import {
+	networkError,
+	getSalt,
+	getVerifiers,
+	genHashDevices,
+} from '../__tests__/constants';
+
+/**
+ *
+ * @param {boolean} success defines if a network request is successful
+ * @param {object?} optional data to return onSuccess, some tests requires specific object values
+ */
+export function netRequestMockSuccess(success, data = {}) {
+	if (success) {
+		jest
+			.spyOn(Client.prototype, 'request')
+			.mockImplementationOnce((...args) => {
+				args[2](null, data);
+			});
```

Comment:
Would you consider this more readable?
up for debate 😅 

```suggestion
			.mockImplementationOnce((...[,,callback]) => {
				callback(null, data);
			});
```

Reply:
javascript is a strange, strange beast